### PR TITLE
Change CI so that it hopefully caches now

### PR DESCRIPTION
**Pull request by thomkeh**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,35 +8,48 @@ on:
...
+    # install dependencies if cache does not exist
+    #----------------------------------------------
     - name: Install dependencies
-      run: poetry install --no-interaction
+      if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+      run: poetry install --no-interaction --no-root
     - name: Format with black
       run: |
+        source .venv/bin/activate
         python -m black --check .
```
Comment (olliethomas) [inline .github/workflows/ci.yml:49]: i think that instead of the activate, you can do 
```suggestion
        poetry run python -m black --check .
```

Comment (olliethomas) [reply .github/workflows/ci.yml:49]: I never use poetry envs, but I'm basing this off [the poetry GH workflow](https://github.com/python-poetry/poetry/blob/master/.github/workflows/main.yml#L86)

Comment (thomkeh) [reply .github/workflows/ci.yml:49]: Let's try it!

**Comment by thomkeh** [comment]
Nice, caching saved 2 minutes for running the CI on the latest commit.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -8,35 +8,47 @@ on:
...
+      if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+      run: poetry install --no-interaction --no-root
     - name: Format with black
       run: |
-        python -m black --check .
+        poetry run python -m black --check .
     - name: Format with isort
       run: |
+        source .venv/bin/activate
         isort --check --diff .
```
Comment (olliethomas) [inline .github/workflows/ci.yml:52]:
```suggestion
        poetry run isort --check --diff .
```

Comment (thomkeh) [inline .github/workflows/ci.yml:52]: The same should actually also work for other entry points:

```suggestion
        poetry run isort --check --diff .
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -49,4 +61,5 @@ jobs:
 #        pyright tests
     - name: Test with pytest
       run: |
+        source .venv/bin/activate
         python -m pytest -vv tests/
```
Comment (olliethomas) [inline .github/workflows/ci.yml:65]:
```suggestion
        poetry run python -m pytest -vv tests/
```
🦜

**Comment by olliethomas** [comment]
> Nice, caching saved 2 minutes for running the CI on the latest commit.

is it worth trying the caching of poetry itself?

**Comment by thomkeh** [comment]
> > Nice, caching saved 2 minutes for running the CI on the latest commit.
> 
> is it worth trying the caching of poetry itself?

Hm, installing poetry takes 13seconds. Not sure that's worth it.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -8,36 +8,47 @@ on:
...
+    #----------------------------------------------
+    - name: Check out repository
+      uses: actions/checkout@v2
     - name: Set up Python 3.7
       uses: actions/setup-python@v2
       with:
         python-version: 3.7
+    #----------------------------------------------
+    #  -----  install & configure poetry  -----
+    #----------------------------------------------
```
[the remark this answers is not in the document]
Comment (thomkeh) [reply .github/workflows/ci.yml:23]: ok, sure

[the remark this answers is not in the document]
Comment (thomkeh) [reply .github/workflows/ci.yml:28]: done

**Pull request merged**